pca.go

```diff
@@ -0,0 +1,63 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// PrincipalComponents returns the k principal component direction vectors of
+// the matrix a, projected score vectors and variance values, where variables
+// are represented in the columns and observations are the rows in the input.
+// The principal component direction vectors are found by singular value
+// decomposition of the input.
+// If no singular value decomposition is possible, pc, proj and sigma are returned
+// nil and ok is returned false. k must not be greater than the number of rows less
+// one or the number of columns in a, or PrincipalComponents will panic.
+func PrincipalComponents(a mat64.Matrix, k int) (pc, proj *mat64.Dense, vars []float64, ok bool) {
```

Comment:
I imagine you could do the same as in covariancematrix, and multiply each row by the square root of the weight?